src/test/java/com/google/api/generator/engine/ast/TernaryExprTest.java

```diff
@@ -41,16 +38,13 @@ public void validTernaryExpr_primitiveType() {
             .setThenExpr(thenExpr)
             .setElseExpr(elseExpr)
             .build();
-    assertThat(ternaryExpr.conditionExpr().type()).isEqualTo(TypeNode.BOOLEAN);
-    assertThat(ternaryExpr.thenExpr().type()).isEqualTo(ternaryExpr.elseExpr().type());
-    assertThat(ternaryExpr.type()).isEqualTo(ternaryExpr.thenExpr().type());
+    assertTrue(ternaryExpr.conditionExpr().type().equals(TypeNode.BOOLEAN));
+    assertEquals(ternaryExpr.thenExpr().type(), ternaryExpr.elseExpr().type());
+    assertEquals(ternaryExpr.type(), TypeNode.INT);
   }
```

Comment:
the assert checks are updated to 
(1) the condition is boolean-typed  
(2) thenExpr/ elseExpr are equal/assignable to each other.
(3) the type of whole expression is correct.